pkg/apis/intevents/v1/topic_validation.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2020 The Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"knative.dev/pkg/apis"
+)
+
+func (t *Topic) Validate(ctx context.Context) *apis.FieldError {
+	return t.Spec.Validate(ctx).ViaField("spec")
+}
+
+func (ts *TopicSpec) Validate(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
+
+	if ts.Topic == "" {
+		errs = errs.Also(
+			apis.ErrMissingField("topic"),
+		)
+	}
+
+	switch ts.PropagationPolicy {
+	case TopicPolicyCreateDelete, TopicPolicyCreateNoDelete, TopicPolicyNoCreateNoDelete:
+	// Valid value.
+
+	default:
+		errs = errs.Also(
+			apis.ErrInvalidValue(ts.PropagationPolicy, "propagationPolicy"),
+		)
+	}
+
+	return errs
+}
+
+func (current *Topic) CheckImmutableFields(ctx context.Context, original *Topic) *apis.FieldError {
+	if original == nil {
+		return nil
+	}
+
+	var errs *apis.FieldError
+
+	// Topic is immutable.
+	if original.Spec.Topic != current.Spec.Topic {
```

Comment:
we need to make other attributes immutable as well. Similar to what we had for PullSubscription.